lib/reek/report/report.rb

```diff
@@ -121,6 +123,41 @@ def total_smell_count_message
     end
 
     #
+    # Displays the status of each file as it is examined
+    # Displays the text report when complete
+    #
+    # @public
+    class ProgressReport < TextReport
```

Comment:
Options are passed down as named arguments to the initializer. See for instance `sort_by_issue_count`.